go/vt/vtgate/planbuilder/operators/projection_pushing.go

```diff
@@ -334,23 +334,50 @@ func splitUnexploredExpression(
 	original := sqlparser.Clone(pe.Original)
 	expr := pe.ColExpr
 
-	var colName *sqlparser.ColName
-	if dt != nil {
-		if !pe.isSameInAndOut(ctx) {
-			panic(vterrors.VT13001("derived table columns must be the same in and out"))
-		}
-		colName = sqlparser.NewColNameWithQualifier(pe.Original.ColumnName(), sqlparser.NewTableName(dt.Alias))
-		ctx.SemTable.CopySemanticInfo(expr, colName)
-	}
-
 	// Get a applyJoinColumn for the current expression.
 	col := join.getJoinColumnFor(ctx, original, expr, false)
-	col.DTColName = colName
 
-	return pushDownSplitJoinCol(col, lhs, pe, alias, rhs)
+	if dt == nil {
+		return pushDownSplitJoinCol(col, lhs, rhs, pe, alias)
+	}
+
+	if !pe.isSameInAndOut(ctx) {
+		panic(vterrors.VT13001("derived table columns must be the same in and out"))
+	}
+	// we are pushing a derived projection through a join. that means that after this rewrite, we are on top of the
+	// derived table divider, and can only see the projected columns, not the underlying expressions
+	colName := sqlparser.NewColNameWithQualifier(pe.Original.ColumnName(), sqlparser.NewTableName(dt.Alias))
+	ctx.SemTable.CopySemanticInfo(expr, colName)
+	col.Original = colName
+	if alias == "" {
+		alias = pe.Original.ColumnName()
+	}
+
+	// Update the left and right child columns and names based on the applyJoinColumn type.
+	switch {
+	case col.IsPureLeft():
+		lhs.add(pe, alias)
+		col.LHSExprs[0].Expr = colName
+	case col.IsPureRight():
+		rhs.add(pe, alias)
+		col.RHSExpr = colName
+	case col.IsMixedLeftAndRight():
```

Comment:
can this just be the default? If the column is not solved by only left or right, then it has to be mixed, right? We already panic in getJoinColumnFor if that is not the case